src/controllers/controller.bar.js

```diff
@@ -1,6 +1,6 @@
 import DatasetController from '../core/core.datasetController';
 import defaults from '../core/core.defaults';
-import Rectangle from '../elements/element.rectangle';
+import {Rectangle} from '../elements/index';
```

Comment:
I haven't tested it, but from reading the docs it sounds like webpack will do a better job of tree shaking at least in some scenarios depending on the options the user has set if we leave the import statement as `import Rectangle from '../elements/element.rectangle';`. It sounds like it has different modes where it either looks at just `import`/`export` or sometimes tries to figure out if code has side-effects although that's tougher

https://webpack.js.org/guides/tree-shaking/

Reply:
I fail to find such information than webpack's guide. Most of the examples import non default exports there.

Reply:
Importing non-default export is fine. I'm more worried about importing things that aren't used. I.e. the existence of `elements/index`. I'm a little less worried about `elements/index` if nothing refers to it though except the entry point and we make a new entry point for tree shaking

Reply:
This should make it possible to have `export * from ./elements` in that new entry point.